File: trunk/Phanfare.ExternalAPI/Section.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Xml;

namespace Phanfare.ExternalAPI
{
	public class Section
	{
		public long AlbumID { get; set; }
		public long SectionID { get; set; }
		public string Name { get; set; }
		public string Description { get; set; }
		public int ImageCount { get; set; }
		public ImageInfo[] Images { get; set; }

		public static Section FromXML( XmlElement el )
		{
			Section section = new Section();
			section.AlbumID = el.GetAttributeInt64( "album_id" );
			section.SectionID = el.GetAttributeInt64( "section_id" );
			section.Name = el.GetAttribute( "section_name" );
			section.Description = el.GetAttribute( "section_descr" );

			XmlElement imagesElement = el[ "images" ];
			int actualCount;
			section.Images = imagesElement.ReadList<ImageInfo>( "num_images", ImageInfo.FromXML, out actualCount );
			section.ImageCount = actualCount;

			return section;
		}
	}
}

File: trunk/PhanfareJavaAPI/src/com/phanfare/api/Section.java
package com.phanfare.api;

import org.json.me.JSONArray;
import org.json.me.JSONException;
import org.json.me.JSONObject;

public class Section {
	public long albumId;
	public long sectionId;
	public String name;
	public String description;
	public int imageCount;
	public ImageInfo[] images;

	public static Section fromJson(JSONObject object) {
		Section section = new Section();
		try {
			section.albumId = object.getLong("album_id");
			section.sectionId = object.getLong("section_id");
			section.name = object.getString("section_name");
			section.description = object.getString("section_descr");
			section.imageCount = object.getInt("num_images");
			JSONArray images = object.optJSONArray("image_infos");
			if (images != null) {
				int length = images.length();
				if (length == section.imageCount) {
					section.images = new ImageInfo[length];
					for (int n = 0; n < section.images.length; n++) {
						section.images[n] = ImageInfo.fromJson(images.getJSONObject(n));
					}
				}
			}
		} catch (JSONException e) {
			// TODO Auto-generated catch block
			e.printStackTrace();
			return null;
		}
		return section;
	}
}
